Description: Set wxInfoBarGeneric text colour correctly
 For some reason SetOwnForegroundColour() was used, but we really need to set
 the colour of the child text control and not our own colour here (which is not
 used for anything).
 This makes the text visible even when using non default tooltip colours, which
 could previously result in black-on-black text.
Origin: upstream, http://trac.wxwidgets.org/changeset/75213
Bug: http://trac.wxwidgets.org/ticket/15671
Applied-Upstream: http://trac.wxwidgets.org/changeset/75213
Last-Update: 2013-11-30

Index: /wxWidgets/branches/WX_3_0_BRANCH/src/generic/infobar.cpp
===================================================================
--- a/src/generic/infobar.cpp (revision 74602)
+++ b/src/generic/infobar.cpp (revision 75213)
@@ -70,7 +70,5 @@
 
     // use special, easy to notice, colours
-    const wxColour colBg = wxSystemSettings::GetColour(wxSYS_COLOUR_INFOBK);
-    SetBackgroundColour(colBg);
-    SetOwnForegroundColour(wxSystemSettings::GetColour(wxSYS_COLOUR_INFOTEXT));
+    SetBackgroundColour(wxSystemSettings::GetColour(wxSYS_COLOUR_INFOBK));
 
     // create the controls: icon, text and the button to dismiss the
@@ -81,4 +79,5 @@
 
     m_text = new wxStaticText(this, wxID_ANY, "");
+    m_text->SetForegroundColour(wxSystemSettings::GetColour(wxSYS_COLOUR_INFOTEXT));
 
     m_button = wxBitmapButton::NewCloseButton(this, wxID_ANY);
